### PR TITLE
♻️ Refactor primary key column types

### LINKED ISSUE #168: ♻️ Get SequelizeConnectionManager through IoC
#### Describe your issue

The [SequelizeConnectionManager](https://github.com/essential-projects/sequelize_connection_manager) was thus far only a simple exported function.
It was now [refactored into a full class with ioc support](https://github.com/essential-projects/sequelize_connection_manager/pull/6).

The sequelize-based repositories need to be refactored, so that they expect the manager to be injected as a dependency.

Affected repositories include:
- [ ] [Correlation](https://github.com/process-engine/correlations.repository.sequelize)
- [ ] [ExternalTask](https://github.com/process-engine/external_task.repository.sequelize)
- [ ] [FlowNodeInstance](https://github.com/process-engine/flow_node_instance.repository.sequelize)
- [ ] [ProcessModels](https://github.com/process-engine/process_model.repository.sequelize)
- [ ] [Timer](https://github.com/process-engine/timers.repository.sequelize)

Each of them needs to get the following additions:
- Update `SequelizeConnectionManager` to 2.0.0
- An additional dependency `SequelizeConnectionManager` to their IoC registrations
- An additional injection parameter for the manager in their constructors

Furthermore, the runtime and all meta repositories with integrationtests need to register the ioc_module of the `SequelizeConnectionManager` in their setup.
This affects the following meta repositories:
- [ ] [Consumer API Meta](https://github.com/process-engine/consumer_api_meta)
- [ ] [Deployment API Meta](https://github.com/process-engine/deployment_api_meta)
- [ ] [ExternalTask API Meta](https://github.com/process-engine/external_task_api_meta)
- [ ] [Management API Meta](https://github.com/process-engine/management_api_meta)
- [ ] [Process Engine Meta](https://github.com/process-engine/process_engine_meta)

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Remove the defined `id` columns from every repository and let sequelize auto-generate that column instead.
    - Our definitions contained uuids for these columns, which doesn't work with every db technology.
2. Add migrations to accomondate these changes
    - Also use a different naming schema for the migrations to prevent odering errors that Umzug appears to be running infos, when dealing with version numbers such as `2.9.0` and `2.10.0`.
3. Update the SequelizeConnectionManager version and add it to the ioc setup.
4. Use `~` type dependencies for selected packages.

**Issues:**

Closes #168 

PR: #170

## How can others test the changes?

All should run as before, since these are purely internal changes.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Fixing Security Issues   | `:lock:`             | 🔒     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️      |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️      |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: sequelize/migrations/flow_node_instance/07_flow_node_instance.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    const flowNodeInstanceIdHasMatchingType = flowNodeInstanceTableInfo.id.type === 'INTEGER';
+    const processTokenIdHasMatchingType = processTokenTableInfo.id.type === 'INTEGER';
+
+    if (flowNodeInstanceIdHasMatchingType && processTokenIdHasMatchingType) {
+      console.log('The database is already up to date. Nothing to do here.');
+      return;
+    }
+
+    if (!processTokenIdHasMatchingType) {
```
Comment (robpal94) [inline sequelize/migrations/flow_node_instance/07_flow_node_instance.js]: Extract the Business Rule here

File: sequelize/migrations/flow_node_instance/07_flow_node_instance.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                                type, payload, flowNodeInstanceId, createdAt, updatedAt
+                            FROM ProcessTokens;`;
+
+      await queryInterface.sequelize.query(updateQuery);
+
+      queryInterface.dropTable('ProcessTokens');
+      queryInterface.renameTable('ProcessTokens_New', 'ProcessTokens');
+    }
+
+    if (!flowNodeInstanceIdHasMatchingType) {
```
Comment (robpal94) [inline sequelize/migrations/flow_node_instance/07_flow_node_instance.js]: and here.

File: sequelize/migrations/flow_node_instance/07_flow_node_instance.js (hunk 1 above)
Comment (ElRaptorus) [reply sequelize/migrations/flow_node_instance/07_flow_node_instance.js]: I think `const !processTokenIdHasNoMatchingType === !processTokenIdHasMatchingType` seems a bit... eccentric. 😉

File: sequelize/migrations/flow_node_instance/07_flow_node_instance.js (hunk 2 above)
Comment (ElRaptorus) [reply sequelize/migrations/flow_node_instance/07_flow_node_instance.js]: see above.

File: sequelize/migrations/flow_node_instance/07_flow_node_instance.js (hunk 1 above)
Comment (robpal94) [reply sequelize/migrations/flow_node_instance/07_flow_node_instance.js]: Hmm, fair point 😉

**Pull request merged**